deepspeed/runtime/zero/stage3.py

```diff
@@ -2828,6 +2699,8 @@ def unscale_and_clip_grads(self, sub_group_id, total_norm):
             clip = ((total_norm / self.loss_scale) + 1e-6) / self.clip_grad
             if clip > 1:
                 combined_scale = clip * self.loss_scale
+        # to maintain behavior of averaging over accumulation steps
+        combined_scale *= self.micro_step_id + 1
```

Comment:
Won't self.micro_step_id be zero at this point because of [this](https://github.com/microsoft/DeepSpeed/blob/e236293bd610ff793a6a1a3bf294b2b6f4b254a3/deepspeed/runtime/zero/stage3.py#L2430) and [this](https://github.com/microsoft/DeepSpeed/blob/e236293bd610ff793a6a1a3bf294b2b6f4b254a3/deepspeed/runtime/zero/stage3.py#L2392)?

Reply:
yep i think youre right. will fix this and add a unit test